pkg/setting/setting.go

```diff
@@ -810,27 +811,13 @@ type RemoteCacheOptions struct {
 }
 
 func (cfg *Cfg) readSessionConfig() {
-	sec := cfg.Raw.Section("session")
-	SessionOptions = session.Options{}
-	SessionOptions.Provider = sec.Key("provider").In("memory", []string{"memory", "file", "redis", "mysql", "postgres", "memcache"})
-	SessionOptions.ProviderConfig = strings.Trim(sec.Key("provider_config").String(), "\" ")
-	SessionOptions.CookieName = sec.Key("cookie_name").MustString("grafana_sess")
-	SessionOptions.CookiePath = AppSubUrl
-	SessionOptions.Secure = sec.Key("cookie_secure").MustBool()
-	SessionOptions.Gclifetime = cfg.Raw.Section("session").Key("gc_interval_time").MustInt64(86400)
-	SessionOptions.Maxlifetime = cfg.Raw.Section("session").Key("session_life_time").MustInt64(86400)
-	SessionOptions.IDLength = 16
-
-	if SessionOptions.Provider == "file" {
-		SessionOptions.ProviderConfig = makeAbsolute(SessionOptions.ProviderConfig, cfg.DataPath)
-		os.MkdirAll(path.Dir(SessionOptions.ProviderConfig), os.ModePerm)
-	}
+	sec, _ := cfg.Raw.GetSection("session")
 
-	if SessionOptions.CookiePath == "" {
-		SessionOptions.CookiePath = "/"
+	if sec != nil {
+		log.Warn("[Removed] Session setting does not exist anymore, use remote_cache option instead")
 	}
 
-	SessionConnMaxLifetime = cfg.Raw.Section("session").Key("conn_max_lifetime").MustInt64(14400)
+	return
```

Comment:
This returns seems pointless. I saw @aocenas did something similar, is there some kind of tooling that adds them? :)

Reply:
Not in my case, I just did not notice it after I reordered some Ifs that it is not needed anymore.